src/Core/Models/Business/SubscriptionUpdate.cs

```diff
@@ -300,13 +302,15 @@ private void AddNewSecretsManagerItems(List<SubscriptionItemOptions> updatedItem
         updatedItems.Add(new SubscriptionItemOptions
         {
             Price = _plan.StripeSeatPlanId,
-            Quantity = _additionalSeats
+            Quantity = _additionalSeats,
+            Deleted = _additionalSeats == 0 ? true : null,
         });
 
         updatedItems.Add(new SubscriptionItemOptions
         {
             Price = _plan.StripeServiceAccountPlanId,
-            Quantity = _additionalServiceAccounts
+            Quantity = _additionalServiceAccounts,
+            Deleted = _additionalServiceAccounts == 0 ? true : null
         });
```

Comment:
It looks like you're using `Deleted` to avoid adding a line item for 0 items at $0.00, is that right? (per my comment here: https://github.com/bitwarden/server/pull/3087#discussion_r1266225013)

`Deleted` is to indicate that the item should be deleted from an existing subscription. But if they're signing up for SM, they won't have this line item yet. Not sure if it'll cause any problems, it just seems like a misuse of the API. (unless I'm missing something)

Why not just check `if (_additionalSeats > 0)` and `if (_additionalServiceAccounts > 0)` before adding the relevant item?

Reply:
you have a good point. Thanks